splash_ingest/handlers.py

```diff
@@ -21,3 +21,9 @@ def __call__(self, point_number, key):
         # print(str(point_number))
         return dask.array.from_array(self._datasets[key])[start:stop][0, :, :] 
         # return self._datasets[key][start:stop][0, :, :]
+
+    def __del__(self):
+        # Explicitly do NOT close on __del__.
+        # This will be fixed upstream in the base class (in PR
```

Comment:
> After the event_model fix is released, will this need to be removed? Is the summary of that fix that is we never need to close h5 files explicitly? That when the file is GC'd, it gets closed auto-magically? Is the close method in h5py really just intended for ending write access?

I think you mean when `area-detector-handlers` is released. https://github.com/bluesky/area-detector-handlers/pull/25 No, this will become superfluous and _may_ be removed but it will do no harm if left as is. Yes, when GC-ed, the files are closed auto-magically. You could argue that `close()` is more important for write (because closing it makes it possible for another process to read from or write to that file). But I think `close()` still matters for read... it's just more convenient to rely on GC in situations where many different parts of the code are using the file, whereas in simple situations like

```py
with h5py.File(..., "r") as file:
    data = file[...]
# close() is called on exit from context
```

there no reason to rely on GC and it's "clean" to explicitly release the file.

Reply:
I am confused. If `HandlerBase.__del__` is not calling `close()` anymore, than where is it the file being closed?

Reply:
It's being closed by the GC whenever the `h5py.File` itself goes out of scope. In this case, that will occur whenever the last `dask.array` that wraps a dataset from that file performs its I/O is replaced by the resultant `numpy.ndarray` within its `xarray.Dataset`.

Reply:
> But I think close() still matters for read... it's just more convenient to rely on GC in situations where many different parts of the code are using the file, whereas in simple situations like

I have made muddied the waters here. In the abstract, I think `close()` is still important to implement and should be used in simple situations like the code snippet above, but this is not a simple situation, so we should rely on the GC.